sys/fmt/fmt.c

```diff
@@ -62,6 +62,17 @@ size_t fmt_byte_hex(char *out, uint8_t byte)
     return 2;
 }
 
+size_t fmt_bytes_hex(char *out, const uint8_t *ptr, size_t n)
+{
+    size_t i = 0;
+    while (i < n) {
```

Comment:
uiuiui. ;)

please replace the while loop with:
```
if (out) {
  while (n--) {
    out += fmt_byte_hex(out, *ptr++);
  }
}
```

(untested, but should be equivalent)

1. all functions in fmt just return the number of bytes that would have written, should ```out``` be ```NULL```
2. while your code might be more readable to some, better keep the style of the rest of the file.